modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala

```diff
@@ -127,29 +128,34 @@ class SeqTranslate(site: Site, systems: Systems, settings: Settings) {
             x.configure(config).map(_ => Result.Configured(x.resource)).toAction(kind)
           },
           List(Action(ActionType.Observe, Kleisli(ctx => observe(config, obsId, inst, sys.filterNot(inst.equals), headers)(ctx).run.map(_.toResult)), Action.State(Action.Idle, Nil))))
-
       extractStatus(config) match {
-        case StepState.Pending => Step.init(
+        case StepState.Pending if i >= nextToRun => Step.init(
```

Comment:
For some reason in my tests the ODB was sending as "ready" steps that should have been "skipped". This workaround takes care of this.